src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+    ptvsd.wait_for_attach()
+    print('debugger attached')
+    return _handler(event, context)
+
+`
+
+        const outFilePath = path.join(params.outputDir, `${debugHandlerFileName}.py`)
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile outFilePath:', outFilePath)
+        await writeFile(outFilePath, template)
+
+        return {
+            outFilePath,
+            debugHandlerName: `${debugHandlerFileName}.${debugHandlerFunctionName}`
+        }
+    } catch (err) {
+        logger.error('makeLambdaDebugFile failed:', err as Error)
+        throw err
+    }
+}
+
+const fixFilePathCapitalization = (filePath: string): string => {
+    if (process.platform === 'win32') {
+        const startsWithLowercase = new RegExp(/^[a-z].*/)
+        if (startsWithLowercase.test(filePath)) {
+            return  filePath.slice(0, 1).toUpperCase() + filePath.slice(1)
+        }
+    }
+
+    return filePath
+}
+
+const makeDebugConfig = ({debugPort, samProjectCodeRoot}: {
+    debugPort?: number,
+    samProjectCodeRoot: string,
+}): PythonDebugConfiguration => {
+    return {
+        type: PYTHON_LANGUAGE,
+        request: 'attach',
+        name: 'SamLocalDebug',
+        host: 'localhost',
+        port: debugPort!,
+        pathMappings: [
+            {
+                // tslint:disable-next-line:no-invalid-template-strings
+                localRoot: fixFilePathCapitalization(samProjectCodeRoot),
+                remoteRoot: '/var/task'
+            }
+        ],
+    }
+}
+
+export async function initialize({
     configuration,
     toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
 
-    const invokeLambda = async (args: LambdaLocalInvokeArguments) => {
+    const invokeLambda = async (args: LambdaLocalInvokeParams) => {
         const samProjectCodeRoot = await getSamProjectDirPathForFile(args.document.uri.fsPath)
-        logger.debug(`Project root: '${samProjectCodeRoot}'`)
+        const baseBuildDir = await makeBuildDir()
+
         let debugPort: number | undefined
 
+        let handlerName: string
+        let manifestPath: string | undefined
         if (args.isDebug) {
             debugPort = await getDebugPort()
+            const {debugHandlerName} = await makeLambdaDebugFile({
+                handlerName: args.handlerName,
+                debugPort: debugPort,
+                outputDir: samProjectCodeRoot,
+            })
+            handlerName = debugHandlerName
+            manifestPath = await makePythonDebugManifest({
+                samProjectCodeRoot,
+                outputDir: baseBuildDir
+            })
+        } else {
+            handlerName = args.handlerName
```

Comment:
Consider initializing handlername with args.handlerName and eliminating the else block

Reply:
✔️